test/testutils.jl

```diff
@@ -1,23 +1,44 @@
-allapprox(coords₁::C, coords₂::C; kwargs...) where {C<:CRS} =
-  all(ntuple(i -> isapprox(getfield(coords₁, i), getfield(coords₂, i); kwargs...), nfields(coords₁)))
-
-function allapprox(coords₁::C, coords₂::C; kwargs...) where {C<:Cartesian2D}
-  isapprox(coords₁.x, coords₂.x, kwargs...) && isapprox(coords₁.y, coords₂.y, kwargs...)
+svec(coords::Cartesian) = SVector(getfield(coords, :coords))
+svec(coords::CoordRefSystems.Projected) = SVector(coords.x, coords.y)
+
+is_approx_angle(
+  α, β;
+  atol = sqrt(eps(2 * convert(promote_type(Unitful.numtype.((α, β))...), π))),
```

Comment:
Could this be simplified in terms of the following?

```julia
α′, β′, π′ = promote(α, β, π)
```

If we can express the comparison without manually playing with units, that would be ideal.

Reply:
Some playing with units is unavoidable as we want to take the `sqrt()` just on the number part, not on the unit. But yes, the trick of using `promote()` instead of `convert(promote_type(numtype(...)))` is quite neat, and we can use it to simplify the computations here to 
```suggestion
  atol = sqrt(eps(2 * ustrip(rad, promote(π, α, β)[1]))),
```